### PR TITLE
convert to typescript 2.0

### PR DESCRIPTION
**Pull request by FredKSchott**
- [X] CHANGELOG.md has been updated (not applicable)

/cc @rictic @usergenic @justinfagnani

### CODE REVIEW

File: custom_typings/winston.d.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import {LeveledLogMethod} from 'winston';
+
+// TODO(fks) 11-10-2016: Remove this once @types/winston has been updated
+// (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12613)
```
Comment (rictic) [inline custom_typings/winston.d.ts]: :+1:

File: lib/index.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
+/**
```
Comment (rictic) [inline lib/index.js]: This file and `lib/index.d.ts` look like accidental includes

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  _logger: LoggerInstance;
+
+  /**
+   * Constructs a new instance of PolymerLogger. This creates a new internal
+   * `winston` logger, which is what we use to handle most of our logging logic.
+   */
+  constructor(options?: {level?: string, name?: string}) {
+    options = options || {};
+
+    this._transport = new (transports.Console)({
```
Comment (rictic) [inline src/index.ts]: Unnecessary parens, `new transports.Console(/* ... */);` is equivalent and easier to read

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+'use strict';
+
+import {Logger, LoggerInstance, ConsoleTransportInstance, transports} from 'winston';
```
Comment (rictic) [inline src/index.ts]: FYI: you need to add a `/// <ref>` for your winston custom typings if you want the d.ts file to be able to reference any of the types therein (which you may not).

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+class PolymerLogger {
+
+  _transport: ConsoleTransportInstance;
+  _logger: LoggerInstance;
+
+  /**
+   * Constructs a new instance of PolymerLogger. This creates a new internal
+   * `winston` logger, which is what we use to handle most of our logging logic.
+   */
+  constructor(options?: {level?: string, name?: string}) {
```
Comment (rictic) [inline src/index.ts]: What are the allowable levels? If possible, explain this using the type system in the winston custom typings, then reference it everywhere here that expects one of these levels, e.g.

```typescript
type Level = 'info' | 'error' | 'warn' | 'debug';
```

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+   * to give the user useful information for debugging purposes. These will
+   * generally only be displayed when the user is are troubleshooting an
+   * issue.
+   */
+  debug(_msg: string, ..._metadata: any[]): void {
+    return this._log.bind(this, 'debug').apply(this, arguments);
+  }
+}
+
+module.exports = {
```
Comment (rictic) [inline src/index.ts]: This style is opaque to typescript I think. Recommend instead:

```typescript
export let level: winston.Level = 'info';
export function setVerbose() { /* ... */};
/* etc */
```

**Comment by FredKSchott** [comment]
Wow this is old, already tackled by @rictic in a separate PR

**Pull request closed without merging**